sdk/dotnet/Pulumi/Pulumi.csproj

```diff
@@ -29,6 +29,10 @@
     <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
   </PropertyGroup>
 
+  <ItemGroup>
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```

Comment:
I think this is a very big reference to add. What functionality is this change giving us? I didn't think any of your Serilog changes required an AspNetCore reference. I think we should be wary of this in the core lib

Reply:
The only thing I can think of is the `using Microsoft.Extensions.Configuration;` that was added in `Deployment.cs`? If that's it I would just take which `Microsoft.Extensions.Configuration.*` dependencies we need directly instead of adding the framework reference.

Reply:
I was trying to avoid referencing a framework dependent version of [Microsoft.Extensions.Configuration.EnvironmentVariables](Microsoft.Extensions.Configuration.EnvironmentVariables).